CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+    $<$<STREQUAL:${CMAKE_GENERATOR},Ninja>:
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/sqlite
+    >
+  PRIVATE
+    src
+    src/soci/src/core
+    src/soci/include/private)
+target_link_libraries (soci
+  NIH::sqlite3
+  Ripple::boost)
+add_library (NIH::soci ALIAS soci)
+target_link_libraries (ripple_libs INTERFACE NIH::soci)
+
+#[===================================================================[
+   NIH dep: snappy
+#]===================================================================]
+
+add_library (snappy STATIC
+  src/snappy/snappy/snappy.cc
+  src/snappy/snappy/snappy-sinksource.cc
+  src/snappy/snappy/snappy-stubs-internal.cc)
+target_compile_options (snappy
+  PRIVATE
+    $<$<NOT:$<BOOL:${MSVC}>>:
+      -Wno-deprecated
+      -Wno-unused-function
+    >
+    $<$<BOOL:${is_gcc}>:-Wno-nonnull-compare>)
+target_include_directories (snappy
+  PUBLIC
+    src/snappy/snappy
+    src/snappy/config)
+add_library (NIH::snappy ALIAS snappy)
+target_link_libraries (ripple_libs INTERFACE NIH::snappy)
+
+#[===================================================================[
+   NIH dep: rocksdb
+#]===================================================================]
+
+include (CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG ("-std=c++11" COMPILER_SUPPORTS_CXX11)
+if (NOT MSVC AND COMPILER_SUPPORTS_CXX11)
+  add_library (rocksdb STATIC
+    #[=========================================================[
+       this unity file is our interpretation of what sources
+       comprise rocksdb (this must be investigated/confirmed
+       whenever updated)
+    #]=========================================================]
+    src/ripple/unity/rocksdb.cpp)
+  target_compile_options (rocksdb
+    PRIVATE
+      $<$<BOOL:${is_gcc}>:-w>
+    PUBLIC
+      $<$<AND:$<BOOL:${is_clang}>,$<VERSION_GREATER_EQUAL:CMAKE_CXX_COMPILER_VERSION,7>>:
+        -Wno-inconsistent-missing-override
+        -Wno-uninitialized
+      >
+      $<$<BOOL:${is_clang}>:--system-header-prefix=rocksdb2>)
+  target_include_directories (rocksdb
+    PRIVATE
+      src
+      src/rocksdb2
+    PUBLIC
+      src/rocksdb2/include)
+  target_compile_definitions (rocksdb
+    PUBLIC
+      SNAPPY
+      RIPPLE_ROCKSDB_AVAILABLE=1
+      $<$<BOOL:${WIN32}>:ROCKSDB_PLATFORM_WINDOWS>
+      $<$<NOT:$<BOOL:${WIN32}>>:
+        ROCKSDB_PLATFORM_POSIX
+        ROCKSDB_LIB_IO_POSIX
+      >
+      $<$<BOOL:${APPLE}>:OS_MACOSX=1>
+      $<$<PLATFORM_ID:Linux>:OS_LINUX>
     )
+  target_link_libraries (rocksdb NIH::snappy)
+else ()
+  add_library (rocksdb INTERFACE)
+  target_compile_definitions (rocksdb INTERFACE RIPPLE_ROCKSDB_AVAILABLE=0)
+endif ()
+add_library (NIH::rocksdb ALIAS rocksdb)
+target_link_libraries (ripple_libs INTERFACE NIH::rocksdb)
+
+#[===================================================================[
+   NIH dep: nudb
+
+   NuDB is header-only, thus is an INTERFACE lib in CMake.
+   TODO: move this into NuDB repo, add proper targets and
```

Comment:
Did you consider contacting the library author and asking him to add exported targets for his or her library?

Reply:
Upstream is https://github.com/vinniefalco/NuDB and Vinnie seems focused on boost/beast these days. NuDB hasn't been touched in over a year even though it has some design issues (https://github.com/vinniefalco/NuDB/issues/46) and some trivial PRs that could be easily merged (https://github.com/vinniefalco/NuDB/pull/62).